File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/performance/datasource/CpuDataSource.java
package com.didichuxing.doraemonkit.kit.performance.datasource;

import com.didichuxing.doraemonkit.kit.performance.PerformanceDataManager;
import com.didichuxing.doraemonkit.kit.performance.widget.LineData;

public class CpuDataSource implements IDataSource {
    @Override
    public LineData createData() {
        float rate = PerformanceDataManager.getInstance().getLastCpuRate();
        return LineData.obtain(rate, Math.round(rate) + "%");
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/performance/manager/datasource/CpuDataSource.kt
package com.didichuxing.doraemonkit.kit.performance.manager.datasource

import com.didichuxing.doraemonkit.kit.performance.manager.PerformanceDataManager
import com.didichuxing.doraemonkit.kit.performance.manager.widget.LineData
import com.didichuxing.doraemonkit.kit.performance.manager.widget.LineData.Companion.obtain
import kotlin.math.roundToInt

/**
 *
 * Desc:cpu数据工厂实现类
 * <p>
 * Date: 2020-06-09
 * Updater:
 * Update Time:
 * Update Comments:
 *
 * Author: pengyushan
 */
class CpuDataSource : IDataSource {
    override fun createData(): LineData? {
        val rate: Float = PerformanceDataManager.instance.lastCpuRate
        return obtain(rate, rate.roundToInt().toString() + "%")
    }
}